ServerCore/Pages/Submissions/Index.cshtml.cs

```diff
@@ -174,88 +177,47 @@ private async Task SetupContext(int puzzleId)
         ///     lockout.
         /// </summary>
         /// <param name="ev"></param>
-        /// <param name="submissions">
-        ///     Expects submissions in chronological order
-        /// </param>
-        /// <param name="puzzleState"></param>
+        /// <param name="WrongSubmissionCount"></param>
         /// <returns>
         ///     If the team should be locked out, returns the time when a team
         ///     can enter submissions again.
         ///     Null if the team should not be locked out.
         /// </returns>
         private static DateTime? ComputeLockoutExpiryTime(
             Event ev,
-            IList<Submission> submissions,
-            PuzzleStatePerTeam puzzleState)
+            int WrongSubmissionCount)
         {
-            int consecutiveWrongSubmissions = 0;
-
-            /**
-             * Count the number of submissions in the past N minutes where N is
-             * the LockoutIncorrectGuessPeriod set for the event. If that count
-             * exceeds the LockoutIncorrectGuessLimit set for the event, then
-             * the team should be locked out of that puzzle.
-             */
-
-            foreach (Submission s in submissions.Reverse())
-            {
-                // Do not increment on partials
-                if (s.Response != null)
-                {
-                    continue;
-                }
-
-                if (s.TimeSubmitted.AddMinutes(ev.LockoutIncorrectGuessPeriod)
-                        .CompareTo(DateTime.UtcNow) < 0)
-                {
-                    break;
-                }
-
-                ++consecutiveWrongSubmissions;
-            }
-
-            if (consecutiveWrongSubmissions <= ev.LockoutIncorrectGuessLimit) {
+            if (WrongSubmissionCount < ev.LockoutIncorrectGuessLimit) {
                 return null;
             }
 
-            /**
-             * The lockout duration is determined by the difference between the
-             * count of wrong submissions in the lockout period and the lockout
-             * limit. That difference is multiplied by the event's 
-             * LockoutDurationMultiplier to determine the lockout time in
-             * minutes.
-             */
-
-            return DateTime.UtcNow.AddMinutes(
-                (consecutiveWrongSubmissions -
-                 ev.LockoutIncorrectGuessLimit) *
-                ev.LockoutDurationMultiplier);
-
+            return DateTime.UtcNow.AddMinutes(Math.Max(
+                ev.LockoutBaseDuration,
```

Comment:
The identifier name LockoutBaseDuration seems misleading, suggesting a base that's added to rather than a maximum.  So I'd suggest renaming this to something like LockoutMaxMinutes.